### PR TITLE
Fixed build of example code to use new binary messenger API.

### PR DESCRIPTION
**Pull request by gaaclarke**
Reference: https://github.com/flutter/flutter/issues/35773

### CODE REVIEW

**Review by jmagman — APPROVED** [review]
LGTM

**Review by jmagman — CHANGES_REQUESTED** [review]
I just pushed https://github.com/flutter/flutter/pull/35778, so now you can test with 
```
SHARD=build_tests dart dev/bots/test.dart
```
Can you remove the local expectIpaBuildFailure variable and the expectIpaBuildFailure parameter from _flutterBuildIpa? https://github.com/flutter/flutter/blob/master/dev/bots/test.dart#L275

**Comment by gaaclarke** [comment]
> I just pushed #35778, so now you can test with
> 
> ```
> SHARD=build_tests dart dev/bots/test.dart
> ```
> 
> Can you remove the local expectIpaBuildFailure variable and the expectIpaBuildFailure parameter from _flutterBuildIpa? https://github.com/flutter/flutter/blob/master/dev/bots/test.dart#L275

Done, can you please test it?  I can't run it locally.

File: dev/bots/test.dart [hunk 1]
```diff
@@ -272,12 +272,10 @@ Future<void> _runBuildTests() async {
       continue;
     }
     final String examplePath = fileEntity.path;
-    final String basename = path.basename(examplePath);
-    final bool expectIpaBuildFailure = basename == 'platform_channel_swift'; // Remove when https://github.com/flutter/flutter/issues/35773 is fixed.
 
     await _flutterBuildAot(examplePath);
     await _flutterBuildApk(examplePath);
-    await _flutterBuildIpa(examplePath, expectIpaBuildFailure: expectIpaBuildFailure);
+    await _flutterBuildIpa(examplePath);
```
Comment (jmagman) [inline dev/bots/test.dart]: Also could you remove the expectIpaBuildFailure parameter in _flutterBuildIpa since now it's unused?

**Comment by jmagman** [comment]
> > I just pushed #35778, so now you can test with
> > ```
> > SHARD=build_tests dart dev/bots/test.dart
> > ```
> > 
> > 
> > Can you remove the local expectIpaBuildFailure variable and the expectIpaBuildFailure parameter from _flutterBuildIpa? https://github.com/flutter/flutter/blob/master/dev/bots/test.dart#L275
> 
> Done, can you please test it? I can't run it locally.

Done, all pass!  https://github.com/flutter/flutter/pull/35825/checks?check_run_id=165935920 will run it too.

File: dev/bots/test.dart (hunk 1 above)
Comment (gaaclarke) [reply dev/bots/test.dart]: Done.

**Review by jmagman — APPROVED** [review]

**Comment by greensopinion** [comment]
FYI the [article on medium](https://medium.com/flutter-community/deep-links-and-flutter-applications-how-to-handle-them-properly-8c9865af9283) needs updating to reflect this change

**Comment by jmagman** [comment]
> FYI the [article on medium](https://medium.com/flutter-community/deep-links-and-flutter-applications-how-to-handle-them-properly-8c9865af9283) needs updating to reflect this change

@DenisovAV

**Pull request merged**